sql/core/src/test/scala/org/apache/spark/sql/execution/joins/BroadcastJoinSuite.scala

```diff
@@ -479,9 +479,9 @@ abstract class BroadcastJoinSuiteBase extends QueryTest with SQLTestUtils
   test("broadcast join where streamed side's output partitioning is PartitioningCollection") {
     withSQLConf(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "500") {
       val t1 = (0 until 100).map(i => (i % 5, i % 13)).toDF("i1", "j1")
-      val t2 = (0 until 100).map(i => (i % 5, i % 13)).toDF("i2", "j2")
```

Comment:
These slight changes are needed to avoid `ReuseExchange` nodes in the test plans as this PR changes the `ReuseExchangeAndSubquery` rule to identify reuses based on their canonicalized plans only (the schema of the exchanges are no longer need to match).
Without these changes the canonicalized plans of `LocalTableScanExec` in t1, t2 and t4 are the same as they have the same content.

Reply:
@peter-toth are you saying that the new logic may reuse an exchange even if the schema doesn't match? The canonicalized form of the plan doesn't consider schema?

Reply:
Schema contains the name of the output attributes as well, but in canonicalized plans we always replace names to "none" and only the structure of plan matters.
Please also note that `ReusedExchangeExec` captures the output of the original exchange so that can we restore the schema in a `ReusedExchangeExec`.

Reply:
I see, so the new logic is more effective (can trigger reuse in more cases), and we need to update the test here to not trigger reuse.

Reply:
Yes it can, but I don't think this comes up frequently in real world applications.
Changing `t2` and `t4` seemed easier than preparing the test for reused exchanged.
I also looked into why the issue didn't come up with the AQE version of the suite (`BroadcastJoinSuiteAE`) before this PR and that's because the test doesn't run the queries before inspecting the plans and the initial AQE plan don't contain reuses (but final plans do).